### DISCUSSION TITLE
some questions about the champsim's statistical data,please help me 😭

### DISCUSSION THREAD

**Discussion by icuuuuuuer** [ChampSim/ChampSim#485] [Q&A]
I'm new to champsim and have some questions about the statistical data.
I configured the L1D cache with a next-line prefetcher, and no other caches have been configured with prefetchers.

The statistical data shows as follows:
**cpu0_L1D TOTAL ACCESS: 229403 HIT: 136977 MISS: 92426
cpu0_L1D LOAD ACCESS: 114138 HIT: 66386 MISS: 47752
cpu0_L1D RFO ACCESS: 0 HIT: 0 MISS: 0
cpu0_L1D PREFETCH ACCESS: 106098 HIT: 61460 MISS: 44638
cpu0_L1D WRITE ACCESS: 8928 HIT: 8927 MISS: 1
cpu0_L1D TRANSLATION ACCESS: 239 HIT: 204 MISS: 35
cpu0_L1D PREFETCH REQUESTED: 114138 ISSUED: 106100 USEFUL: 2375 USELESS: 808

cpu0_L1I TOTAL ACCESS: 20 HIT: 5 MISS: 15
cpu0_L1I LOAD ACCESS: 20 HIT: 5 MISS: 15
cpu0_L2C PREFETCH ACCESS: 3193 HIT: 3 MISS: 3190
cpu0_L2C PREFETCH REQUESTED: 0 ISSUED: 0 USEFUL: 0 USELESS: 0**

based on the issue https://github.com/ChampSim/ChampSim/issues/170,I understand that " Since a prefetch hit in the L1D by an L1D prefetch does not cause a change in state, a prefetch hit in the L1D is not counted in the useful/useless metrics. Only the L1D prefetch misses, which fills the L1D with a new cache line, are measured for usefulness."
My questions are:
1.The sum of USEFUL and USELESS in cpu0_L1D PREFETCH REQUESTED is significantly less than the value of MISS in cpu0_L1D PREFETCH. Why is this the case? Shouldn't they add up to be strictly equal?
2.Why does the cpu0_L2C PREFETCH ACCESS have a value of 3193 here? My L2 Cache is not configured with a prefetcher. I can understand that it might be a prefetch initiated by L1D, but I can't seem to find a sum of values in L1D that equals the number 3193.

Does anyone know the reason? Thank you very much!

**Comment by maccoymerrell** [comment]
1. I believe useless/useful statistics are only updated when a prefetched cache line is evicted. Useful/useless will be only the prefetched lines that are brought into the cache and subsequently evicted. The quality of a prefetch cannot be determined before this.
2. If I had to guess, 3193 ~= 2375 + 808. Why they don't exactly match may be due to the state of the simulation upon ending. If thats the case, then tallying may be done on prefetches evicted to the L2 from the L1, but not misses from the L1. That is 'probably' a bug.

**Reply by icuuuuuuer** [reply]
Thank u very much,it's very clear,very helpful!
